lib/readline.js

```diff
@@ -1007,10 +1054,7 @@ Interface.prototype._ttyWrite = function(s, key) {
         // falls through
 
       default:
-        if (s instanceof Buffer)
-          s = s.toString('utf-8');
-
-        if (s) {
+        if (typeof s === 'string' && s) {
```

Comment:
Why this change? Is this code path never going to get buffers?

Reply:
It could get `null` and empty strings.

Reply:
`null` when triggering specials keys, see https://nodejs.org/api/readline.html#readline_rl_write_data_key

Empty strings are send when `readline` is used by `repl`, removing `&& s` check fails editor test for `repl`. Also it'll make it redraw the line unnecessarily.

Reply:
@silverwind at least no core code would ever call this function with a buffer. It could theoretically be called directly by a user with a buffer. That would be surprising though and it would also be the only code path in the whole function that supports buffers. All others would not handle them and therefore it did not seem right to check for buffers (it was a legacy check from ~6-7 years ago).

Reply:
Okay, sounds fine.